### PR TITLE
Remove spawn III

### PR DESCRIPTION
**Pull request by coiax**
I feel like these PRs are of differing lengths.

### CODE REVIEW

File: code/datums/mind.dm [hunk 1]
```diff
@@ -1718,8 +1718,7 @@
 			if(istype(S, type))
 				continue
 		S.charge_counter = delay
-		spawn(0)
-			S.start_recharge()
+		addtimer(S, "start_recharge", 0)
```
Comment (MrStonedOne) [inline code/datums/mind.dm]: Spawns like these should be replaced with set waitfor = 0 procs, as the spawn is about avoiding the sleep in the proc its calling

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -218,9 +215,7 @@
 				if(D.operating)
 					D.nextstate = OPEN
 				else if(D.density)
-					spawn(0)
-					D.open()
-	return
```
Comment (MrStonedOne) [inline code/game/area/areas.dm]: this spawn didn't actually do anything, kek

**Comment by Core0verload** [comment]
What benefits `addtimer` has over `spawn`, besides added complexity and proc overhead?

**Comment by optimumtact** [comment]
We can control it better because it's using a subsystem we can set priorities on

File: code/datums/mind.dm (hunk 1 above)
Comment (optimumtact) [inline code/datums/mind.dm]: was this addressed? Can't quite tell form the diff

**Comment by coiax** [comment]
You say "proc overhead" but actually a spawn is anywhere between twice and six times as expensive as an addtimer call

**Comment by MrStonedOne** [comment]
So far the only downside is that the per-run overhead can get expensive if there are lots of things in the queue. I'll be fixing that with my next mc related pr

File: code/datums/mind.dm (hunk 1 above)
Comment (MrStonedOne) [inline code/datums/mind.dm]: I made a pr to make addtimer(..,..,0) basically equate to a set waitfor = 0 call of the proc.

**Comment by optimumtact** [comment]
Can't wait for everything to break!

**Pull request merged**